eng/_util/golangci/go.mod

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+module github.com/microsoft/go/_util/golangci
```

Comment:
Why a new go.mod? That somewhat defeats the point of `go tool [x]` IMO.

Reply:
I'm ok leaving this out for now (just split out the lint fixes and merge them so they don't get stale) until @qmuntal has a chance to take a look at this decision.

Reply:
Perhaps it's enough to use the latest Docker container image and just make sure the raw image sha/digest shows up in logs, so that it *could* be reproduced locally in the unexpected case that it makes a difference to which errors are found by CI vs. locally. Pinning dependencies is my knee-jerk reaction to seeing a new one pop up, but in this case it's "just" a linter, it *should* have no breaking changes, it has no security privilege (right?), and the maintainers actively suggest not to build it from source.